inst/tests/tests.Rraw

```diff
@@ -6921,10 +6921,23 @@ ll = sapply(ll, paste, collapse=",")
 test(1477.07, transpose(strsplit(ll, ",", fixed=TRUE)), tstrsplit(ll, ",", fixed=TRUE))
 test(1477.08, transpose(1:5), error="l must be a list")
 test(1477.09, transpose(list(as.complex(c(1, 1+5i)))), error="Unsupported column type")
-test(1477.10, transpose(list(list(1:5))), error="Item 1 of list input is")
+test(1477.10, transpose(list(x~y)), error="Item 1 of list input is") # changed behavior since we also support now list columns #5805
 test(1477.11, transpose(as.list(1:5), fill=1:2), error="fill must be a length 1 vector")
 test(1477.12, transpose(as.list(1:5), ignore.empty=NA), error="ignore.empty should be logical TRUE/FALSE")
 test(1477.13, transpose(list()), list())
+# return list columns #5639
+la = list(as.list(1:3), list("a","b","c"))
+lb = list(list(1L,"a"), list(2L,"b"), list(3L,"c"))
+test(1477.21, transpose(list(1:3, c("a","b","c")), list.cols=TRUE), lb)
+test(1477.22, transpose(list(1:3, c("a","b","c")), list.cols=FALSE), lapply(lb, as.character))
```

Comment:
nit: `unlist()` is probably closer to what actually happens? since `unlist()` is more commonly associated with type promotion...

```suggestion
test(1477.22, transpose(list(1:3, c("a","b","c")), list.cols=FALSE), lapply(lb, unlist))
```